### PR TITLE
Removed boost/any in favour of openvdb::MetaMap usage

### PR DESCRIPTION
**Pull request by Idclip**

### CODE REVIEW

File: openvdb/points/PointDataGrid.h [hunk 1]
```diff
@@ -1163,73 +1163,103 @@ PointDataLeafNode<T, Log2Dim>::readBuffers(std::istream& is, bool fromHalf)
...
+        return *(it->value());
+    }
+    else {
+        typename StreamT::Ptr pagedStream = std::make_shared<StreamT>();
+        (const_cast<MetaMap&>(auxData)).insertMeta(key, TypedMetaT(pagedStream));
+        return *pagedStream;
+    }
+}
+
+} // namespace page_stream
```
Comment (Idclip) [inline openvdb/points/PointDataGrid.h:1216]: @danrbailey be good for you to take a look through this - I started splitting out all the descriptor methods from the `Local` structs into a shared namespace too, but there's ever so slightly different implementations of some. Would be good to consolidate them

File: openvdb/points/PointDataGrid.h [hunk 2]
```diff
@@ -1163,73 +1163,103 @@ PointDataLeafNode<T, Log2Dim>::readBuffers(std::istream& is, bool fromHalf)
...
+
+template <typename StreamT>
+inline StreamT&
+getOrInsertPagedStream(const MetaMap& auxData,
+                       const Index index)
+{
+    using TypedMetaT = TypedMetadata<typename StreamT::Ptr>;
+
+    const std::string key("paged:" + std::to_string(index));
+    typename TypedMetaT::ConstPtr it = auxData.getMetadata<TypedMetaT>(key);
```
Comment (Idclip) [inline openvdb/points/PointDataGrid.h:1205]: note - `getMetadata` performs string type checks. If we don't want to do that, we can use `static_cast<...>(auxData[key])`

**Comment by Idclip** [comment]
@danrbailey bump

**Review by danrbailey — APPROVED** [review]

**Comment by danrbailey** [comment]
This PR has now been superceded by this one (https://github.com/AcademySoftwareFoundation/openvdb/pull/1687), closing...

**Pull request closed without merging**